truth_set/python/structure/solution-approach/standard-method/OrderProcessor.py

```diff
@@ -0,0 +1,19 @@
+class OrderProcessor:
+    def apply_discounts(self, order_total, discounts_applied):
+        if discounts_applied > 5:
+            raise Exception("Too many discounts applied.")
+            
+        if order_total > 100:
+            order_total -= 10  
+        elif order_total > 50:
+            order_total -= 5
+            
+        return order_total
+
+processor = OrderProcessor()
+
+try:
+    final_total = processor.apply_discounts(120, 6)
+    print(f"Final Total: ${final_total}")
+except Exception as error:
+    print(f"Error: {str(error)}")
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider separating test and implementation code</b></div>
<div id="fix">

Consider moving the test code into a separate test file. The instantiation and testing of `OrderProcessor` should not be part of the class implementation file. A similar issue was also found in truth_set/python/structure/solution-approach/standard-method/OrderProcessor.py (line 13-19).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion

```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/1#issuecomment-2536353661>#c621b6</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged